interfaces/builtin/qualcomm_ipc_router.go

```diff
@@ -79,22 +273,52 @@ func (iface *qualcomIPCRouterInterface) BeforeConnectPlug(plug *interfaces.Conne
 		// then the host system doesn't have the required feature to compile the
 		// policy, the qipcrtr socket is a new addition not present in i.e.
 		// xenial
-		return fmt.Errorf("cannot connect plug on system without qipcrtr socket support")
+		return fmt.Errorf("cannot %s on system without qipcrtr socket support", what)
 	}
 
 	return nil
 }
 
+func (iface *qualcomIPCRouterInterface) BeforePrepareSlot(slot *snap.SlotInfo) error {
+	if slot.Attrs == nil {
+		slot.Attrs = make(map[string]interface{})
+	}
+	if isSlotInfoSystem(slot) {
+		// qcipc defaults to interface name for system slots
+		slot.Attrs[qcipcAttrib] = iface.Name()
+		return nil
+	}
+
+	qcipc, ok := slot.Attrs[qcipcAttrib].(string)
+	if !ok {
+		return fmt.Errorf("qualcomm-ipc-router slot must have a %s attribute", qcipcAttrib)
+	}
+	if err := validateTag(qcipc); err != nil {
+		return err
+	}
+
+	address, ok := slot.Attrs[addressAttrib].(string)
+	if !ok {
+		return fmt.Errorf("qualcomm-ipc-router slot must have an %s attribute", addressAttrib)
+	}
+	if err := validateAddress(address); err != nil {
+		return err
+	}
+
+	return iface.verifySupport("prepare slot")
+}
+
+func (iface *qualcomIPCRouterInterface) BeforeConnectPlug(plug *interfaces.ConnectedPlug) error {
+	var qcipc string
+	if err := plug.Attr(qcipcAttrib, &qcipc); err != nil {
+		// In no qcipc attribute, this plug is probably meant to
+		// connect to the system slot: we make sure the attribute will
+		// match (so checks specified by allow-connection are passed).
+		plug.SetAttr(qcipcAttrib, iface.Name())
```

Comment:
we don't really want to do this, also usually the fallback would be the name of the plug but given the backward compat constraint we can't have that either